ingestion-beam/src/test/java/com/mozilla/telemetry/SinkMainTest.java

```diff
@@ -0,0 +1,109 @@
+
+import static com.mozilla.telemetry.matchers.Lines.matchesInAnyOrder;
+import static org.junit.Assert.assertThat;
+
+import com.google.common.io.Resources;
+import com.mozilla.telemetry.matchers.Lines;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline.PipelineExecutionException;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class SinkMainTest {
+
+  @Rule
+  public TemporaryFolder outputFolder = new TemporaryFolder();
+
+  private String outputPath;
+
+  @Before
+  public void initialize() {
+    outputPath = outputFolder.getRoot().getAbsolutePath();
+  }
+
+  @Test
+  public void canWriteToStdout() {
+    String input = Resources.getResource("testdata/single-message-input.json").getPath();
+
+    // We are simply making sure this runs without throwing an exception.
+    Sink.main(new String[]{
+        "--inputFileFormat=json",
+        "--inputType=file",
+        "--input=" + input,
+        "--outputFileFormat=json",
+        "--outputType=stdout"
+    });
+  }
+
+  @Test(expected = PipelineExecutionException.class)
+  public void throwsExceptionOnNullPlaceholder() {
+    String input = Resources.getResource("testdata/single-message-input.json").getPath();
+
+    Sink.main(new String[]{
+        "--inputFileFormat=json",
+        "--inputType=file",
+        "--input=" + input,
+        "--outputFileFormat=json",
+        "--outputType=file",
+        "--output=tmp/${some_nonexistent_attribute}/out"
+    });
+  }
+
+  @Test
+  public void canWriteSingleRecord() {
+    String input = Resources.getResource("testdata/single-message-input.json").getPath();
+    String output = outputPath + "/out";
+
+    // We are simply making sure this runs without throwing an exception.
+    Sink.main(new String[]{
+        "--inputFileFormat=json",
+        "--inputType=file",
+        "--input=" + input,
+        "--outputFileFormat=json",
+        "--outputType=file",
+        "--output=" + output
+    });
+
+    List<String> inputLines = Lines.files(input);
+    List<String> outputLines = Lines.files(outputPath + "/*.ndjson");
+
+    assertThat(outputLines, matchesInAnyOrder(inputLines));
+  }
+
+  @Test
+  public void canWriteErrorOutput() {
+    String input = Resources.getResource("testdata/basic-messages-mixed-validity.ndjson").getPath();
+    String output = outputPath + "/valid/out";
+    String errorOutput = outputPath + "/error/out";
+
+    // We are simply making sure this runs without throwing an exception.
+    Sink.main(new String[]{
+        "--inputFileFormat=json",
+        "--inputType=file",
+        "--input=" + input,
+        "--outputFileFormat=json",
+        "--outputType=file",
+        "--output=" + output,
+        "--errorOutputType=file",
+        "--errorOutput=" + errorOutput
+    });
+
+    List<String> validLines = Lines.resources("testdata/basic-messages-normalized.ndjson");
```

Comment:
for consistent naming this file should be `testdata/basic-messages-valid.ndjson`